app/javascript/styles/mastodon/components.scss

```diff
@@ -279,11 +279,12 @@
     display: inline-flex;
     align-items: center;
     width: auto !important;
+    padding: 0 4px 0 2px;
   }
 
   &__counter {
     display: inline-block;
-    width: 14px;
+    width: auto;
```

Comment:
I think this was the `width: auto !important` above the padding you added...

Reply:
The difference is that the class `.icon-button__counter` is the number itself which had the fixed with and the outer button with the class `.icon-button--with-counter` has the `width: auto !important` which just sets the container element to be as wide as the content (including the fixed-width number element)